### PR TITLE
Create task-service module

### PR DESCRIPTION
**Pull request by josephperrott**

### CODE REVIEW

File: shared/github.ts [hunk 1]
```diff
@@ -35,3 +35,19 @@ export interface GithubPullRequestEvent {
...
+/** Pushes a status to github for a commmit. */
+export async function setStatusOnGithub(
+  org: string, repo: string, sha: string, state: 'error' | 'failure' | 'pending' | 'success', 
+  description: string = '', target_url: string = '', context: string = 'branch-manager') {
+  const url = `https://api.github.com/repos/${org}/${repo}/statuses/${sha}`;
+  const config = {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+      'Authorization': 'token <insert token here>',
```
Comment (devversion) [inline shared/github.ts:49]: Is that intended to be left here?

**Comment by devversion** [comment]
So this is another module next to `firestore-module` right? so right now the workflow is:

1) Github webhook --> Update Firestore PR entry
2) Firestore-events module --> on pull request document change -> Trigger `task-service` module
3) Task service module -> Schedules http request using Cloud tasks
4) Presubmit module triggered -> replies with "whether can cherry pick" as http response

Is that right? I think we should definitely have some Google drawing for the workflow, and add it to the readme. In order to be able to review in a helpful way, I want to understand how it is planned 😄 

Previously I just thought that Step 2) just calls Step 4) and the `firestore-events` module handles that. It makes sense to abstract that somehow, but I feel like `tasks-service` and `firestore-events` could be combined, otherwise we pass so many http requests around?

File: shared/github.ts [hunk 2]
```diff
@@ -35,3 +35,19 @@ export interface GithubPullRequestEvent {
...
+    },
+    body: JSON.stringify({state, target_url, description, context})
+  };
+  await fetch(url, config).then(response => response.text()).then(console.log);
```
Comment (jelbourn) [inline shared/github.ts]: Add a `TODO` where things are left to be filled in?

**Review by jelbourn — APPROVED** [review]
LGTM

File: shared/github.ts (hunk 1 above)
Comment (josephperrott) [reply shared/github.ts:49]: I'll add a TODO for loading the token at runtime.

**Comment by josephperrott** [comment]
@DevVersion After our conversation about single threads in firebase functions I realized that we need to split out the cherrypick checks into their own tasks.

I will work on getting a drawing set up that outlines it but overall

1.  Github sends webhooks for events.
2.  Firebase functions handles requests and:
  * Update firestore
  * Determine based on changes what PRs need to be checked and create task for each
3.  Task Service handles each of these tasks that were created, in parallel.


So it goes
```
Github ──> Firebase Functions ─┬──> Firestore
                               └──> Task Service ──> Github

```

**Comment by josephperrott** [comment]
e

**Comment by devversion** [comment]
@josephperrott Thanks Joey. This makes sense. A drawing would be really cool, so that we can review based on that plan/concept that you've created.

**Pull request merged**